### DISCUSSION TITLE
How to write NetCDF with time as type 'int' not 'int64'

### DISCUSSION THREAD

**Discussion by mankoff** [pydata/xarray#5429] [Q&A] [answered]
I'm writing a NetCDF file and checking for [CF compliance](
http://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/cf-conventions.html). I am getting the following error:

```
§2.2 Data Types
* The variable time failed because the datatype is int64
```

Is there some way to make xarray write non-int64 time dimension?

**Comment by spencerkclark** [comment, marked as answer]
Yes, you may set the `"dtype"` encoding attribute of the time coordinate before writing it out to a file.  E.g. this will encode the dates with a `double` precision datatype:
```python
ds.time.encoding["dtype"] = "float64"
ds.to_netcdf("float64-time-coordinate.nc")
```

**Reply by mankoff** [reply]
Thank you - that works.

**Reply by spencerkclark** [reply]
Oh and if you want to encode it as an `int` you can use `"int32"` as the dtype (sorry I just noticed the title of your post).
